src/codegen/dispatch/mod.rs

```diff
@@ -0,0 +1,17 @@
+use super::{cfg::ControlFlowGraph, Options};
+use crate::{sema::ast::Namespace, Target};
+
+pub(super) mod solana;
+pub(super) mod substrate;
+
+pub(super) fn function_dispatch(
+    contract_no: usize,
+    all_cfg: &[ControlFlowGraph],
+    ns: &mut Namespace,
+    opt: &Options,
+) -> ControlFlowGraph {
+    match &ns.target {
+        Target::Solana => solana::function_dispatch(contract_no, all_cfg, ns, opt),
+        _ => substrate::function_dispatch(contract_no, all_cfg, ns, opt),
```

Comment:
Done, however I need to give it to the EVM target though, otherwise we'll have failing tests. In theory this dispatcher _might_ actually even work for the EVM target as well (don't count on it though :grin: ).

Reply:
@xermicus You undid your changes in this file. I am sure I have seen `Target::Substrate` here.

Reply:
Good catch! I made a mistake while pasting the SPDX header, and then I messed up again by doing one undo too much when fixing it :facepalm: Thanks